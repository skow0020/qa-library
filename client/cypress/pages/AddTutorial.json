{
    "title": "#title",
    "category": "#category",
    "language": "#language",
    "url": "#url",
    "backgroundImage": "#backgroundImage",
    "body": "#body",
    "description": "#body"
}