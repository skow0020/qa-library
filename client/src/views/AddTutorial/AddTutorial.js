/* eslint jsx-a11y/anchor-is-valid: 0 */

import AlertModal, { showAlert } from "components/common/AlertModal";
import {
  Button,
  Container,
  Form,
  FormInput,
  FormTextarea,
  Row
} from "shards-react";

import CategoriesSelection from "components/common/CategoriesSelection";
import LanguagesSelection from "components/common/LanguagesSelection";
import PageTitle from "components/common/PageTitle";
import React from "react";
import axios from "axios";

class AddTutorial extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: 0,
      backgroundImage: '',
      category: 'General',
      url: '',
      title: '',
      body: '',
      language: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleCategoryChange = this.handleCategoryChange.bind(this);
    this.handleLanguageChange = this.handleLanguageChange.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleCategoryChange(e) {
    this.setState({ category: e.target.value });
  }

  handleLanguageChange(e) {
    this.setState({ language: e.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const tutorial = {
      backgroundImage: this.state.backgroundImage,
      category: this.state.category,
      url: this.state.url,
      title: this.state.title,
      body: this.state.body,
      language: this.state.language
    };

    axios.post("/api/tutorials/", tutorial)
      .then(
        response => {
          if (response.data.success) { this.showNotifier("Tutorial added successfully"); }
          else { this.showNotifier(`Unable to add tutorial: ${JSON.stringify(response.data.error.errors)}`); }
          this.nextPath('/tutorials');
        },
        error => this.showNotifier(`Unable to add tutorial: ${error}`)
      );
  }

  nextPath(path) {
    setTimeout(function () {
      this.props.history.push(path);
    }.bind(this), 2000);
  }

  showNotifier = (message) => {
    showAlert({ message });
  }

  render() {
    const { url, title, backgroundImage, body, category, language } = this.state;

    return (
      <Container fluid className="main-content-container px-4">
        <AlertModal />
        <Form id="add-tutorial-form" onSubmit={this.handleSubmit}>
          <Row noGutters className="page-header py-4">
            <PageTitle sm="8" title="Add a Tutorial" className="text-sm-left" />
            <Button id="submit" type="submit" className="btn btn-success btn-lg">
              Submit
            </Button>
          </Row>

          <label className="text-muted font-weight-bold" htmlFor="title">Tutorial Title</label>
          <FormInput id="title" type="text" value={title} onChange={this.handleChange} required />
          <Row form>
            <LanguagesSelection id="language" value={language} onChange={this.handleLanguageChange} />
            <CategoriesSelection id="category" value={category} onChange={this.handleCategoryChange} required />
          </Row>
          <label className="text-muted font-weight-bold" htmlFor="url">Tutorial URL</label>
          <FormInput id="url" type="text" value={url} onChange={this.handleChange} required />
          <label className="text-muted font-weight-bold" htmlFor="backgroundImage">Background Image URL</label>
          <FormInput id="backgroundImage" type="text" value={backgroundImage} onChange={this.handleChange} required />
          <label className="text-muted font-weight-bold" htmlFor="body">Tutorial Description</label>
          <FormTextarea id="body" type="textarea" style={{ height: "150px" }} value={body} onChange={this.handleChange} />
        </Form>
      </Container>
    );
  }
}

export default AddTutorial;
