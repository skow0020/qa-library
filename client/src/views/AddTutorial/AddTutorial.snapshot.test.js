import AddTutorial from './AddTutorial';
import React from 'react';
import renderer from 'react-test-renderer';

describe('AddTutorial Snapshot Tests', () => {
  test('AddTutorial snapshot', () => {
    const component = renderer.create(
      <AddTutorial></AddTutorial>
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});