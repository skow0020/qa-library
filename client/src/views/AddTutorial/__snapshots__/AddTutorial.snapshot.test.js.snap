// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTutorial Snapshot Tests AddTutorial snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div />
  <div
    className="makeStyles-paper-2"
  >
    <form
      className="makeStyles-form-3"
      id="add-article-form"
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <h3
          className="MuiTypography-root page-title MuiTypography-h3 MuiTypography-gutterBottom"
        >
          Add a Tutorial
        </h3>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="title"
              id="title-label"
            >
              Title
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textField-152 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="title"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-215 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-216"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 MuiGrid-grid-lg-3"
        >
          <div
            className="MuiFormControl-root makeStyles-formControl-217"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="language-selection"
            >
              Language
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby=" language"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                id="language"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                onAnimationStart={[Function]}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 MuiGrid-grid-lg-3"
        >
          <div
            className="MuiFormControl-root makeStyles-formControl-254"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="category"
            >
              Category
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby=" category"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                id="category"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                General
              </div>
              <input
                onAnimationStart={[Function]}
                type="hidden"
                value="General"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="url"
              id="url-label"
            >
              URL
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textField-152 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="url"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-215 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-216"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="backgroundImage"
              id="backgroundImage-label"
            >
              Background Image
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textField-152 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="backgroundImage"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-215 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-216"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="body"
              id="body-label"
            >
              Description
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textField-152 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
              onClick={[Function]}
            >
              <textarea
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                disabled={false}
                id="body"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                rows={2}
                style={
                  Object {
                    "height": undefined,
                    "overflow": null,
                  }
                }
                value=""
              />
              <textarea
                aria-hidden={true}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                readOnly={true}
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "visibility": "hidden",
                  }
                }
                tabIndex={-1}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-215 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-216"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-1 MuiButton-fullWidth"
        disabled={false}
        id="submit-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </form>
  </div>
</main>
`;
